build: Bump minimum Qt version to 5.5.1

Sjors opened the pull request:
Fixes #13478

Compiled and lightly tested on 10.14.3 against QT 5.12.0.

Sjors committed:
Bump minimum Qt version to 5.5.1

luke-jr commented:
Is there an actual benefit to this?

Has anyone done a check for what versions come with current stable distros?

MarcoFalke commented:
* xenial https://packages.ubuntu.com/xenial/libs/qt5-default
* centos 7 http://mirror.centos.org/centos/7/os/x86_64/Packages/qt5-qtbase-5.9.2-3.el7.x86_64.rpm

Also, 5.5.1 is long EOL anyway and it'd be better if the user built qt from our depends.

fanquake commented:
Concept ACK

fanquake commented on build-aux/m4/bitcoin_qt.m4:
```diff
@@ -116,24 +116,7 @@ AC_DEFUN([BITCOIN_QT_CONFIGURE],[
-    if test "x$bitcoin_cv_need_acc_widget" = xyes; then
-      _BITCOIN_QT_CHECK_STATIC_PLUGINS([Q_IMPORT_PLUGIN(AccessibleFactory)], [-lqtaccessiblewidgets])
-    fi
+    _BITCOIN_QT_CHECK_STATIC_PLUGINS([Q_IMPORT_PLUGIN(AccessibleFactory)], [-lqtaccessiblewidgets])
```
Pretty sure you need to drop this line as well.

Sjors replied:
Indeed, `bitcoin_cv_need_acc_widget` would have been `xno` before.

fanquake commented on doc/dependencies.md:
```diff
@@ -23,7 +23,7 @@ These are the dependencies currently used by Bitcoin Core. You can find instruct
 | Python (tests) |  | [3.4](https://www.python.org/downloads) |  |  |  |
 | qrencode | [3.4.4](https://fukuchi.org/works/qrencode) |  | No |  |  |
-| Qt | [5.9.7](https://download.qt.io/official_releases/qt/) | [5.2](https://github.com/bitcoin/bitcoin/pull/14725) | No |  |  |
+| Qt | [5.9.7](https://download.qt.io/official_releases/qt/) | 5.5.1 | No |  |  |
```
Could you (assuming it is merged) link to this PR as a reference? It's useful when referring to the docs to be able to jump straight to the PR/discussion that bumped a dependency.

jonasschnelli commented:
Concept ACK

Sjors commented:
@luke-jr wrote:
> Is there an actual benefit to this?

See discussion in #13478

I added a link to that discussion in doc/dependencies.md and dropped the `Q_IMPORT_PLUGIN(AccessibleFactory)` line.

practicalswift commented on src/qt/test/apptests.cpp:
```diff
@@ -27,9 +27,7 @@
 #include <QTextEdit>
 #include <QtGlobal>
-#if QT_VERSION >= 0x050000
 #include <QtTest/QtTestWidgets>
```
Could sort includes :-)

flack commented on build-aux/m4/bitcoin_qt.m4:
```diff
@@ -276,7 +258,7 @@ AC_DEFUN([_BITCOIN_QT_CHECK_QT5],[
     ]],
     [[
-      #if QT_VERSION < 0x050200 || QT_VERSION_MAJOR < 5
+      #if QT_VERSION < 0x050501 || QT_VERSION_MAJOR < 5
```
is the `QT_VERSION_MAJOR` clause still needed? At first glance, it looks like a leftover from when Qt 4 was still supported

Sjors replied:
I was assuming this is a generic check to see if QT >=5.5.1 is present at all.

flack replied:
according to https://doc.qt.io/qt-5/qtglobal.html#QT_VERSION if `QT_MAJOR_VERSION` is 4, then `QT_VERSION` should be something like `0x040102`, so it would already fail the first clause. IOW, I don't think there is any constellation where the first clause is false and the second one is true

MarcoFalke replied:
Could ask @TheCharlatan if there was a reason to do this in 28482efefb15dc2c273a52b96d1aceb995db4968

Sjors replied:
@flack I agree the `|| QT_VERSION_MAJOR < 5` bit seems redundant. That maybe have been in order to support both 4.x and 5.x from certain minor versions onward.

TheCharlatan replied:
I was confused about what the actual supported minor version was at the time. Either way, it is definitely redundant with this pr.

fanquake commented:
@Sjors Could you add release notes to this PR as well?

Sjors commented:
Added release note. The gitian binaries should still be valid.

Sjors commented:
I removed the redundant `|| QT_VERSION_MAJOR < 5`.

MarcoFalke commented:
There was a discussion on IRC that debian 8 comes with 5.3 (https://packages.debian.org/jessie/qt5-default)

The conclusion was that anyone on debian 8 uses out `./depends` qt package: http://www.erisian.com.au/bitcoin-core-dev/log-2019-02-14.html#l-405

hebasto commented:
utACK fd46c4c0018c41d36cd892ccb47485b572d65837

MarcoFalke commented:
utACK fd46c4c0018c41d36cd892ccb47485b572d65837

lucayepa commented:
Ok, I'm late to the party, but we are going to have a release that will not build both on Debian Jessie (QT 5.3.2) and Ubuntu Trusty (QT 5.2.1).

I agree with @jonasschnelli :
> <jonasschnelli> "but we need to expect users run it close to the LTS end".

The LTS end date of Ubuntu Trusty is (April 2019)[https://wiki.ubuntu.com/Releases]. The LTS end date of Debian Jessie is (June 30, 2020)[https://wiki.debian.org/LTS].

IMO we should have a policy not to break compatibility with LTS distributions, unless this is really something we need (i.e. bugs or big issues).

Sjors commented:
@lucayepa it's better to bring this up in #13478.

MarcoFalke commented:
@lucayepa It hasn't been building on those distros  before this pull request because no one cared (found it worthwhile) to fix the code

Trusty will be EOL before the final release and if you are on oldstable debian, you can build from depends or use the deterministic binaries.
